convokit/model/corpusComponent.py

```diff
@@ -1,49 +1,152 @@
+from random import randrange
+
 from .convoKitMeta import ConvoKitMeta
 from convokit.util import warn, deprecation
-from typing import List, Optional
+from typing import List, Optional, Type
+
+from convokit.storage import DBDocumentMapping, StorageManager
 
 
 class CorpusComponent:
+    def __init__(self,
+                 obj_type: str,
+                 owner=None,
+                 id=None,
+                 vectors: List[str] = None,
+                 meta=None,
+                 storage: Optional[StorageManager] = None,
+                 from_db=False):
+
+        if storage is not None:
+            self.storage = storage
+        elif owner is not None:
+            self.storage = owner.storage
+        else:
+            self.storage = StorageManager('mem')
+            self.storage.setup_collections(None, None, None, None)
+
+        if from_db:
+            return
+
+        if id is None and obj_type != 'conversation':
+            id = randrange(0, 100)
+
+        if ((obj_type == 'utterance' and id in self.storage._utterances)
+                or (obj_type == 'speaker' and id in self.storage._speakers)):
+            id = f'{id}.1'
+
+        self.fields = self.storage.ItemMapping(
+            self.storage._speakers
+            if obj_type == 'speaker' else self.storage._conversations
+            if obj_type == 'conversation' else self.storage._utterances, id)
```

Comment:
The nested chain of single-line ifs is hard to read; I would factor this out into a traditional if-elif-else switch:
```suggestion
        if obj_type == 'speaker':
            mapped_item = self.storage._speakers
        elif obj_type == 'conversation':
            mapped_item = self.storage._conversations
        else:
            mapped_item = self.storage._utterances
        self.fields = self.storage.ItemMapping(mapped_item, id)
```